tools/generators/epic-generator/index.ts

```diff
@@ -0,0 +1,186 @@
+
+  return convergencePackages;
+};
+
+const getCodeowners = (): Ownership[] => {
+  const codeownersContent = readFileSync('.github/CODEOWNERS', 'utf8');
+
+  return codeownersContent.split('\n').map(line => {
+    const [path, ...owners] = line.split(' ');
+
+    return {
+      path: path.replace(/\/+$/, ''),
+      owners,
+    };
+  });
+};
+
+const mapOwnerships = (packages: Package[], ownerships: Ownership[]): Package[] => {
+  return packages.map(pckg => {
+    const owners = ownerships.find(ownership => ownership.path === pckg.folder)?.owners || [];
+
+    return {
+      ...pckg,
+      owners,
+    };
+  });
+};
+
+const getPackages = (tree: Tree) => {
+  const packages = getConvergencePackages(tree);
+
+  const ownerships = getCodeowners();
+
+  return mapOwnerships(packages, ownerships);
+};
+
+const createEpic = (repo: string, title: string, message: string) => {
+  const issueUrl = execSync(`gh issue create --repo "${repo}" --title "${title}" --body "${message}"`)
+    .toString()
+    .trim();
+
+  return issueUrl;
+};
+
+const createIssue = (repo: string, issue: MigrationIssue, templateTitle: string) => {
+  const title = `${templateTitle} - ${issue.assignee || 'unknown'}`;
+  const message = stripIndents`
+    🚧 This is an auto-generated issue to individually track migration progress.
+
+    ### Packages to migrate:
+    ${issue.packages.map(pckg => `- ${pckg.name}`).join('\n')}
+  `;
+
+  let command = `gh issue create --repo "${repo}" --title "${title}" --body "${message}"`;
+
+  const issueUrl = execSync(command).toString().trim();
+
+  return issueUrl;
+};
+
+const generateIssues = (repo: string, templateTitle: string, packages: Package[]) => {
+  const migrationIssues = packages.reduce<MigrationIssueMap>((acc, pckg) => {
+    const teamOwner = pckg.owners!.find(owner => owner.startsWith('@microsoft/'));
+    const key = teamOwner || 'ownerless';
+
+    if (key in acc) {
+      acc[key].packages.push(pckg);
+    } else {
+      acc[key] = {
+        assignee: teamOwner,
+        packages: [pckg],
+      };
+    }
+
+    return acc;
+  }, {});
+
+  Object.entries(migrationIssues).forEach(([key, issue]) => {
+    migrationIssues[key].issueUrl = createIssue(repo, issue, templateTitle);
+  });
+
+  return migrationIssues;
+};
+
+const updateEpicWithIssues = (epicUrl: string, issueMap: MigrationIssueMap, message: string) => {
+  const packageList = Object.values(issueMap)
+    .map(issue => {
+      return stripIndents`
+    - [ ] ${issue.issueUrl}
+    ${issue.packages.map(pckg => `  - ${pckg.name}`).join('\n')}
+    `;
+    })
+    .join('\n');
+
+  const updatedMessage = stripIndents`
+  ${message}
+
+  ### Packages that need migration:
+  ${packageList}
+`;
+
+  let command = `gh issue edit ${epicUrl} --body "${updatedMessage}"`;
+
+  execSync(command);
+};
+
+export default async function (tree: Tree, schema: EpicGeneratorSchema) {
+  const { title, message, repository } = validateSchema(schema);
+
+  checkAuthentication();
```

Comment:
```suggestion
  await checkAuthentication();
```

should this be awaited ?

Reply:
Nope, for some reason made `checkAuthentication` async and forgot to remove it 🤔 Good catch!